active-directory-new-domain/azuredeploy.json

```diff
@@ -265,32 +155,32 @@
       }
     },
     {
-      "name": "[variables('adNicName')]",
       "type": "Microsoft.Network/networkInterfaces",
+      "name": "[variables('adNicName')]",
+      "apiVersion": "2016-03-30",
       "location": "[resourceGroup().location]",
       "dependsOn": [
         "Microsoft.Resources/deployments/VNet",
-        "[resourceId('Microsoft.Network/loadBalancers/',variables('adLBName'))]"
+        "[resourceId('Microsoft.Network/loadBalancers/', variables('adLBName'))]"
       ],
-      "apiVersion": "2016-03-30",
       "properties": {
         "ipConfigurations": [
           {
             "name": "ipconfig1",
             "properties": {
               "privateIPAllocationMethod": "Static",
-              "privateIPAddress": "[parameters('adNicIPAddress')]",
+              "privateIPAddress": "[variables('adNicIPAddress')]",
               "subnet": {
-                "id": "[variables('adSubnetRef')]"
+                "id": "[concat(resourceId('Microsoft.Network/virtualNetworks', variables('virtualNetworkName')), '/subnets/', variables('adSubnetName'))]"
```

Comment:
No because it's constructing a new name based on the resourceId of the resource it belongs to